# Tarea de la semana 3: Reporte de Datos sobre Felicidad
### Diana Chaglla
### 2023-10-10

En este reporte se presenta estadística descriptiva sobre el nivel de felicidad de varios países del mundo para el primer y último año incluido en el set de datos y se realiza una comparación de valores con dos países seleccionados.

Se empieza importando la librería `pandas` para la manipulación de la base de datos de la semana del 2019-08-07. Los datos fueron extraídos de github usando la función `read_csv()` de `pandas`.

In [9]:
import pandas as pd

In [3]:
data_felicidad = pd.read_csv("https://raw.githubusercontent.com/cienciadedatos/datos-de-miercoles/master/datos/2019/2019-08-07/felicidad.csv")

In [41]:
display(data_felicidad.head())

pais  anio  escalera_vida   log_pib  soporte_social  \
0  Afghanistán  2008       3.723590  7.168690        0.450662   
1  Afghanistán  2009       4.401778  7.333790        0.552308   
2  Afghanistán  2010       4.758381  7.386629        0.539075   
3  Afghanistán  2011       3.831719  7.415019        0.521104   
4  Afghanistán  2012       3.782938  7.517126        0.520637   

   expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0         50.799999  0.718114     0.177889               0.881686   
1         51.200001  0.678896     0.200178               0.850035   
2         51.599998  0.600127     0.134353               0.706766   
3         51.919998  0.495901     0.172137               0.731109   
4         52.240002  0.530935     0.244273               0.775620   

   afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0         0.517637         0.258195   0.612072           -1.929690   
1         0.583926         0.237092   0.611545           -2.044093   
2         0.618265         0.275324   0.299357           -1.991810   
3         0.611387         0.267175   0.307386           -1.919018   
4         0.710385         0.267919   0.435440           -1.842996   

   calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0        -1.655084               1.774662                 NaN   
1        -1.635025               1.722688                 NaN   
2        -1.617176               1.878622                 NaN   
3        -1.616221               1.785360                 NaN   
4        -1.404078               1.798283                 NaN   

   gini_banco_mundial_promedio  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN

El conjunto de datos tiene 17 variables de varios paises del mundo, se estará usando la variable _escalera_vida_ que es un puntaje que indica percepción de calidad de vida en una escala de 0 a 10, siendo el 10 el máximo puntaje.

Para encontrar el primer y último año en el conjunto de datos se usan las funciones `min()` y `max()` en la columna _anio_. 


In [13]:
#Medida de tendencia central y dispersión de la felicidad a nivel mundial para el primer y último año incluido en el set de datos.
#find min and max year
min_a = data_felicidad['anio'].min()
max_a = data_felicidad['anio'].max()
print("El primer año es: " + str(min_a))
print("El último año es: " + str(max_a))

El primer año es: 2005
El último año es: 2018


Conociendo el primer y último año de registro del conjunto de datos se usa la información como condicional para filtar el dataframe y presentar un análisis descriptivo de los datos usando la función `describe()`. Estos conjuntos de datos serán utilizados para análisis posteriores.

In [17]:
felicidad_min = data_felicidad[data_felicidad["anio"] == min_a]
print("Datos_2005")
print(felicidad_min["escalera_vida"].describe())

Datos_2005
count    27.000000
mean      6.446164
std       0.919143
min       4.718734
25%       5.539227
50%       6.619550
75%       7.161203
max       8.018934
Name: escalera_vida, dtype: float64


En el año 2005 existen 27 países, su media es 6.45 y su desviación estándar es 0.92. Es importante notar que el puntaje más bajo es 4.72 y el más alto 8.02.

In [18]:
felicidad_max = data_felicidad[data_felicidad["anio"] == max_a]
print("Datos_2018")
print(felicidad_max["escalera_vida"].describe())

Datos_2018
count    136.000000
mean       5.502134
std        1.103461
min        2.694303
25%        4.721326
50%        5.468088
75%        6.277691
max        7.858107
Name: escalera_vida, dtype: float64


En comparación con el año 2005, la cantidad de países de los que se tiene información es mayor siendo 136 en total con un promedio de 5.50 y desviación estándar de 1.10. Adicionalmente, el mínimo y el máximo descendieron siendo, en el año 2008, 2.69 y 7.85 respectivamente, a su vez la media también descendió aproximandamente un punto.   

Los países que fueron seleccionados para realizar el estudio comparativo son: Venezuela y Australia, y fueron escogidos arbitrariamente. 

Para el año 2005 tenemos los siguientes puntajes en ambos países.

In [28]:
Venezuela_2005 = felicidad_min[felicidad_min["pais"] == "Venezuela"]
print("La felicidad en Venezuela 2005 es: " + Venezuela_2005["escalera_vida"].to_string(index=False))
Australia_2005 = felicidad_min[felicidad_min["pais"] == "Australia"]
print("La felicidad en Australia 2005 es: " + Australia_2005["escalera_vida"].to_string(index=False))

La felicidad en Venezuela 2005 es: 7.169621
La felicidad en Australia 2005 es: 7.340688


Los puntajes de felicidad de Venezuela y Australia en el año 2005 están por encima de la media del conjunto de datos y son bastante similares entre sí.

In [30]:
Venezuela_2018 = felicidad_max[felicidad_max["pais"] == "Venezuela"]
print("La felicidad en Venezuela 2018 es: " + Venezuela_2018["escalera_vida"].to_string(index=False))
Australia_2018 = felicidad_max[felicidad_max["pais"] == "Australia"]
print("La felicidad en Australia 2018 es: " + Australia_2018["escalera_vida"].to_string(index=False))

La felicidad en Venezuela 2018 es: 5.005663
La felicidad en Australia 2018 es: 7.176993


Para el año 2018 el puntaje de felicidad en Venezuela desciende aproximadamente 2 puntos y ahora se encuentra por debajo de la media. Por otro lado, Australia mantiene su puntaje y está muy cercano al valor máximo en ese año.

Para conocer brevemente las posibles variables que pueden influir en el puntaje de felicidad, se realizó un agrupamiento por país en la data de los años previamente mencionados tomando en consideración la información de las columnas _percepcion_corrupcion_, _soporte_social_ y _confianza_.

La variable soporte social es el promedio nacional de las respuestas binarias 0 o 1 que indica si las personas tienen alguien con quien contar en situaciones difíciles, la variable percepción de la corrupción es el promedio nacional entre las respuestas binarias de si consideran que la corrupción esta extendida a nivel gubernamental y la variable confianza es el promedio nacional de las respuestas binarias de si confian o no en el gobierno. 

Se uso la función `groupby()` para el agrupamiento y las función `mean()` para conocer el promedio de las variables en cada país.

In [38]:
mean_2005 = felicidad_min.groupby('pais')[["escalera_vida", "percepcion_corrupcion", "soporte_social", "confianza"]].mean()
print("Promedio 2005:")
display(mean_2005)

Promedio 2005:


escalera_vida  percepcion_corrupcion  soporte_social  \
pais                                                                    
Alemania              6.619550               0.781007        0.963490   
Arabia Saudita        7.079644               0.505149        0.867819   
Australia             7.340688               0.390416        0.967892   
Brasil                6.636771               0.744994        0.882923   
Bélgica               7.262290               0.597554        0.934875   
Canadá                7.418048               0.502681        0.961552   
Dinamarca             8.018934               0.236522        0.972372   
Egipto                5.167754                    NaN        0.847842   
España                7.152786               0.777272        0.961043   
Francia               7.093393               0.687851        0.940338   
Greece                6.006310               0.860563        0.836539   
Hungría               5.193933               0.902811        0.929628   
Iran                  5.308190               0.636490        0.765978   
Italia                6.853784               0.943912        0.928001   
Japón                 6.515817               0.698930        0.927712   
Jordania              6.294660               0.669727        0.920013   
Líbano                5.491245               0.945177        0.796278   
México                6.580658               0.764249        0.902808   
Pakistan              5.224658               0.844436        0.590946   
Países Bajos          7.463979               0.571342        0.947358   
Polonia               5.587209               0.982931        0.921528   
Reino Unido           6.983557               0.398457        0.978840   
República Checa       6.439257               0.900733        0.918759   
Rumania               5.048648               0.956885        0.837685   
Suecia                7.376316                    NaN        0.951470   
Turquía               4.718734               0.876999        0.819936   
Venezuela             7.169621               0.719800        0.955278   

                 confianza  
pais                        
Alemania          0.321759  
Arabia Saudita         NaN  
Australia         0.532063  
Brasil            0.340625  
Bélgica           0.551705  
Canadá            0.442818  
Dinamarca         0.668683  
Egipto                 NaN  
España            0.527846  
Francia           0.324884  
Greece            0.487543  
Hungría           0.360526  
Iran              0.554805  
Italia            0.238738  
Japón             0.350331  
Jordania          0.800150  
Líbano                 NaN  
México            0.426359  
Pakistan          0.583483  
Países Bajos      0.429007  
Polonia           0.068769  
Reino Unido       0.494750  
República Checa   0.282911  
Rumania           0.272887  
Suecia            0.440849  
Turquía           0.595774  
Venezuela         0.491368

En el año 2005 la percepción de la corrupción en Venezuela era bastante alta en comparación a Dinamarca que es el país con mayor puntaje de felicidad, de igual forma la confianza en el gobierno es menor en Venezuela. Sin embargo, el promedio de soporte social es similar en ambos países.

In [40]:
mean_2018 = felicidad_max.groupby('pais')[["escalera_vida", "percepcion_corrupcion", "soporte_social", "confianza"]].mean()
print("Promedio 2018:")
pd.set_option('display.max_rows', None)
display(mean_2018)

Promedio 2018:


escalera_vida  percepcion_corrupcion  soporte_social  \
pais                                                                            
Afghanistán                   2.694303               0.927606        0.507516   
Albania                       5.004403               0.899129        0.683592   
Alemania                      7.118364               0.495674        0.919763   
Algeria                       5.043086               0.758704        0.798651   
Arabia Saudita                6.356393                    NaN        0.867848   
Argentina                     5.792797               0.855255        0.899912   
Armenia                       5.062449               0.676826        0.814449   
Australia                     7.176993               0.404647        0.940137   
Austria                       7.396002               0.523061        0.911668   
Azerbaijan                    5.167995               0.561206        0.781230   
Bangladesh                    4.499217               0.701421        0.705556   
Benin                         5.819827               0.746511        0.503544   
Bielorrusia                   5.233770               0.718455        0.904569   
Bolivia                       5.915734               0.786045        0.827159   
Bosnia y Herzegovina          5.887401               0.912858        0.835890   
Botswana                      3.461366               0.806945        0.794936   
Brasil                        6.190922               0.763251        0.881505   
Bulgaria                      5.098814               0.952014        0.923853   
Burkina Faso                  4.927236               0.757399        0.664859   
Burundi                       3.775283               0.598608        0.484715   
Bélgica                       6.892172               0.630412        0.929816   
Cambodia                      5.121838                    NaN        0.794605   
Camerún                       5.250738               0.884442        0.676825   
Canadá                        7.175497               0.371741        0.922719   
Chad                          4.486325               0.762879        0.577254   
Chile                         6.436221               0.816297        0.890085   
China                         5.131434                    NaN        0.787605   
Chipre                        6.276443               0.848337        0.825573   
Colombia                      5.983512               0.854821        0.870970   
Comoros                       3.972820               0.793758        0.621303   
Congo (Brazzaville)           5.490214               0.738020        0.620623   
Corea del Sur                 5.840231               0.796826        0.797724   
Costa Rica                    7.141075               0.781302        0.875872   
Costa de Marfil               5.268375               0.790967        0.620883   
Croacia                       5.536271               0.925408        0.909807   
Dinamarca                     7.648786               0.150607        0.958219   
Ecuador                       6.128010               0.830743        0.851345   
Egipto                        4.005451                    NaN        0.758824   
El Salvador                   6.276246               0.826600        0.823999   
Emiratos Árabes Unidos        6.603744                    NaN        0.851041   
Eslovaquia                    6.235111               0.909945        0.922379   
Eslovenia                     6.249419               0.839253        0.940971   
España                        6.513371               0.776504        0.910315   
Estados Unidos                6.882685               0.709928        0.903856   
Estonia                       6.091302               0.620678        0.932694   
Etiopía                       4.379262               0.799466        0.740155   
Filipinas                     5.869173               0.726483        0.845803   
Finlandia                     7.858107               0.1

Para el año 2018 el país con mayor puntaje de felicidad es Finlandia que tiene un puntaje bajo en percepción de la corrupción y un nivel medio en confianza en el gobierno, de igual manera Australia tiene niveles similares a Finlandia en las 3 variables analisadas. Venezuela en cambio tiene un nivel muy alto en percepción de la corrupción y muy bajo en la confianza en al gobierno, además el soporte social descendió aproximademente 0.7 en los 13 años de diferencia que hay entre 2005 y 2018. 

A niveles generales se puede concluir que la similitud principal que existe en los países con un puntaje bajo de felicidad es que desconfían del gobierno y consideran que la corrupción esta impregnada en el sistema político. Por lo que se puede concluir que las decisiones gubernamentales infieren en la calidad de vida del pueblo que gobiernan. Sin embargo, para comprobar esta suposición se recomienda hacer análisis más detallados basados en la información de la base de datos.